### PR TITLE
Unit tests for recommender engine, post request 

### PR DESCRIPTION
**Pull request by jasnaurbancic**
**Enhancements:**

- unit tests for recommender engine
- post request to request recommendations

### CODE REVIEW

File: src/server/recsys/test/recommender-tests.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+let x5recommend = require('../engine/x5recommend.js');
+let path = require('path');
+
+const qm = require('qminer');
+
+// internal modules
+const Logger = require('../../../lib/logging-handler')();
+
+// create a logger instance for logging recommendation requests
+const logger = Logger.createGroupInstance('recommendation-requests', 'x5recommend');
```
Comment (ErikNovak) [inline src/server/recsys/test/recommender-tests.js]: Would be better to name 'recommendation-requests-tests'?

File: src/server/recsys/test/recommender-tests.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    let expected = {"error": "Empty query object"};    
+    let recommendations = recsys.recommend({});
+    assert.deepEqual(expected, recommendations);
+    done();
+  });
+  
+  describe('Text queries.', function(){ // describe L2
+    
+    it('Should handle an empty text query', function(done){
+      let expected = {"error": "Empty query object"};
```
Comment (ErikNovak) [inline src/server/recsys/test/recommender-tests.js]: Single quotes

File: src/server/recsys/test/recommender-tests.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      assert.ok(recommendations);
+      done();
+    });
+    
+  }); //describe L2
+
+  describe('URL queries.', function(){ // describe L2
+    
+    it('Should handle an empty url query', function(done){
+      let expected = {"error": "Empty query object"}
```
Comment (ErikNovak) [inline src/server/recsys/test/recommender-tests.js]: Single quotes

File: src/server/recsys/test/recommender-tests.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      let query = { url: '' };
+      let recommendations = recsys.recommend(query);
+      assert.deepEqual(expected, recommendations);
+      done();
+    });
+    
+    it('Should handle a non-existing url query', function(done){
+      let expected = [];
+      let query = { url: 'http://videolectures.net/kdd2016_broder_deep/'}
```
Comment (ErikNovak) [inline src/server/recsys/test/recommender-tests.js]: Semi-colon at end of line

File: src/server/recsys/test/recommender-tests.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      assert.ok(recommendations);
+      done();
+    });
+    
+  }); // describe L2
+  
+  describe('Text and URL queries.', function(){ // describe L2
+    
+    it('Should handle an empty text and url query', function(done){
+      let expected = {"error": "Empty query object"};
```
Comment (ErikNovak) [inline src/server/recsys/test/recommender-tests.js]: Single quotes

File: src/server/recsys/test/recommender-tests.js (hunk 1 above)
Comment (ErikNovak) [reply src/server/recsys/test/recommender-tests.js]: I see. I checked and the log file name is already used [here](https://github.com/JozefStefanInstitute/x5gon/blob/master/src/server/recsys/engine/x5recommend.js#L15). Using the same file name would then (probably) merge both logs from unit tests and from methods within the `x5recommend` class. 

I suggest to rename `recommendation-requests` to `recommendation-tests`, if you agree.

**Comment by jasnaurbancic** [comment]
At the moment src/server/recsys/test/server-tests.js should be run as sudo and recommender engine server should not be running at the same time. This is expected to be fixed with different environments (e.g. production, development, test). Also I added supertest to dependencies in package.json.

**Pull request merged**